### ISSUE TITLE
[Suggestion] Make "Freeze chat scrolling" a multi-select dropdown (+other cleanup suggestions)

### ISSUE THREAD

**Issue by abbie** [FrankerFaceZ/FrankerFaceZ#284]
**Priority**: Low

**Web Browser**: All

**Do you use BetterTTV or other Twitch extensions**: No

**FFZ Logs (via FFZ Menu > About > Logs; if Applicable)**: n/a

**Bug / Idea**: Change the `Freeze chat scrolling` dropdown (*Chat>Behavior*) to a multi-select dropdown to remove unnecessary repetition and allow for more user customization. The current dropdown format looks pretty out of place when the rest of the features and settings are done so nicely. Also use this style for all future options that present similar situations. Sloppy example: [WEBM](https://comfy.moe/dyawou.webm) GIF:
![](https://i.imgur.com/cLS87SL.gif)

Things I imagine being different compared to my example:
* Keep the width of the selection box static.
* Make the selection box at the top display hotkey names only to reduce clutter. An example for all options selected would be `On hover, ALT, CTRL, META, SHIFT`. The dropdown selections would still display "When <key> is held" to reduce confusion.
* Possibly change selected options background color instead of text color.

I'm not super familiar with JS or CSS or I would of done these things myself. I just wanted to make this example really quick to show the idea.

Possible alternatives that would still clean up the setting:

**Sub-options**
- Checked, sub-options visible:
![](https://comfy.moe/geqbmt.png)

- Unchecked, sub-options hidden:
![](https://comfy.moe/hfyyvy.png)

**Dropdown + key selection**
![](https://comfy.moe/dovoze.png)
![](https://comfy.moe/usihnq.png)
![](https://comfy.moe/nutxnr.png)

**Option groups (still pretty messy)**
![](https://comfy.moe/bvxgdg.png)

Full options menu images of the above in case you want to see how any of them look with the surroundings:
- [Sub-options Checked](https://comfy.moe/lwwywz.png), [Sub-options Unchecked](https://comfy.moe/gkgquc.png)
- [Key selection Hover](https://comfy.moe/oqcift.png), [Key selection Key](https://comfy.moe/wvtqja.png), [Key selection Both](https://comfy.moe/iujwky.png)
- [Option groups](https://comfy.moe/wnpqov.png)

(Also I think the sub-options style would work very well for `Stream uptime` (*Channel>Metadata*) with a sub checkbox for "Show seconds" as a replacement to the dropdown.) 

**Steps to Reproduce (if Applicable)**: n/a

**Comment by SirStendec** [comment]
First off, I want to say that I absolutely love this issue. Thanks for caring about the UX so much and coming up with all this. You're absolutely correct that the current setting I added is low tier and could be done so much better.

There are a few reasons I did it the way I did. First, it's basically just a straight port of the old setting, which only had support for a list with no multi-select or anything fancy at all. Second, I still haven't finished the settings UI system so it'd be problematic to do things the ways you suggested.

Most importantly, it was 3:30am and I just really wanted to get the setting out there and available to people to use.

I'm going to be going back and changing it, if no one else ends up tackling settings UI before I get to it, but for now my priority is getting more of the core features working. Eventually though, I do plan on letting you pick an arbitrary key with a key picker along with a checkbox for mouse movement.

It'd be a combination of your sub-options suggestion along with a key input field.

**Comment by abbie** [comment]
Yeah I get why this isn't something that should be of top priority right now. Just wanted to give my feedback on it. I think the sub-options + key input will work well. Thanks for the quick response, and hope you don't have too much trouble trying to get everything else done.
